src/sources/util/http/prelude.rs

```diff
@@ -152,13 +152,20 @@ pub trait HttpSource: Clone + Send + Sync + 'static {
 
             info!(message = "Building HTTP server.", address = %address);
 
-            let listener = tls.bind(&address).await.unwrap();
```

Comment:
I think we'd actually just want to change this line to be:

```suggestion
            let listener = tls.bind(&address).await?;
```

potentially with a `.map_err(...)` if the compiler isn't happy about the error type.

If a function/method already has a return type of `Result<T, E>`, it doesn't _generally_ make sense to handle errors yourself and then return dummy values for `E` instead.

Reply:
The return error type in this function is `()`, so I didn't want to lose the error type during `.map_err(...)` from `TlsError` to `()`. But if you say so - I will do it.

Reply:
Fixed.

Reply:
Oh right, duh.

Sorry, I wasn't reading closely enough that this is _inside_ the future we return, where the error type is indeed `()`. I was focusing on the `crate::Result<...>` return type for `run` where it's `Box<dyn Error>`.

Alright, so now that I've mentally caught up here (😅) I think what you originally had was the best we could reasonably do here. Ideally we'd want it to try and bind in `run` to return the error earlier, but that's probably a bit too big of a change to figure out here.

Apologies for the back-and-forth.

Reply:
No worries, I've reverted the changes. Ready for the review again :)